analyses/cell-type-wilms-tumor-14/scripts/utils/00_preprocessing_rds_functions.R

```diff
@@ -0,0 +1,75 @@
+library(dplyr)
+library(Seurat)
+library(ggpubr)
+pre_seuratobj <- function(obj, nfeatures = 500, run_harmony = TRUE, reduction = "harmony", ndims = 50,
+                          skip_logNorm = TRUE){
+  ######## Normalize, scale, feature selection
+  if (skip_logNorm){
+    obj[["RNA"]]$data <- obj[["RNA"]]$counts
+  }else{
+    obj <- Seurat::NormalizeData(obj, normalization.method = "LogNormalize")
+  }
+  obj <- Seurat::FindVariableFeatures(obj, selection.method = "vst", nfeatures = nfeatures)
+  obj <- Seurat::ScaleData(obj, features = Seurat::VariableFeatures(object = obj))
+  # obj <- Seurat::SCTransform(obj)
+  obj <- Seurat::RunPCA(obj, features = Seurat::VariableFeatures(object = obj))
+  # Seurat::ElbowPlot(obj, ndims = 50)
+  
+  ######## batch effect correction
+  if (run_harmony){
+    obj <- harmony::RunHarmony(obj, group.by.vars = "library_id")
+  }
+  
+  ######## Clustering and dimentional reduction
+  obj <- Seurat::FindNeighbors(obj, dims = 1:ndims, reduction = reduction)
+  obj <- Seurat::FindClusters(obj, resolution = 0.8, algorithm = 1)
+  obj <- Seurat::RunUMAP(obj, dims = 1:ndims, reduction = reduction)
+  
+  return(obj)
+}
+
+## create and process a seurat object from provided sce object
+process_sce <- function( sample, library,
+                         path_proj,
+                         db_proj,
+                         path_anal,
+                         get_logcounts = TRUE
+){
+  
+  ########### per sample rds preprocessing ##########
+  
+  #sample <- "SCPCS000517"; library <- "SCPCL000849"
+  rds <- readRDS( file.path(path_proj, sample, paste0(library,"_processed.rds")) )
+  db <- read.table(file.path(db_proj, sample, paste0(library,"_processed_scdblfinder.tsv")), header = T)
+  
+
+  ######## annotate doublets
+  #rds <- rds[,which(db$class == "singlet")]
+  rds$class <- db$class
+  
+  ######## create seurat object from the SCE counts matrix
+  if (get_logcounts) {
+    seurat_obj <- SeuratObject::CreateSeuratObject(counts = SingleCellExperiment::logcounts(rds),assay = "RNA",project = library)
+  }else {
+    seurat_obj <- SeuratObject::CreateSeuratObject(counts = SingleCellExperiment::counts(rds),assay = "RNA",project = library)
+  }
```

Comment:
I think you want to keep the raw counts either way, so maybe:

(Using the SeuratObject function to allow it to check or reformat if desired)

```suggestion
  seurat_obj <- SeuratObject::CreateSeuratObject(counts = SingleCellExperiment::counts(rds),assay = "RNA",project = library)
  
  # add normalized data if requested
  if (get_logcounts) {
    SeuratObject::LayerData(seurat_obj[["RNA"]], "data") <- SingleCellExperiment::logcounts(rds)
  }
```